### PR TITLE
TF-2202 Fix sending email failed when network is corrupted but wifi still in full

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2202

## Resolved

- Establish connection: 
  + Wifi `Lamoda`: Has internet connection
  + Wifi `OPPO F7`: Personal hotspot no network connection

https://github.com/linagora/tmail-flutter/assets/80730648/de04f781-d2dd-47dd-a03f-714825ed7165

### CODE REVIEW

**Comment by hoangdat** [comment]
Can we also display the network banner base on also the internet checker?

**Review by sherlockvn — APPROVED** [review]

**Comment by dab246** [comment]
> Can we also display the network banner base on also the internet checker?

That's right. The banner is displayed when one of the two conditions `connectivityResult == ConnectivityResult.none` or `internetConnectionStatus == InternetConnectionStatus.disconnected` is met.

File: lib/features/network_connection/presentation/network_connection_controller.dart [hunk 1]
```diff
@@ -4,73 +4,109 @@ import 'package:connectivity_plus/connectivity_plus.dart';
...
+    internetConnectionStatus.value = newStatus;
+  }
+
   bool isNetworkConnectionAvailable() {
-    return connectivityResult.value != ConnectivityResult.none;
+    if (PlatformInfo.isWeb) {
+      return connectivityResult.value != ConnectivityResult.none;
+    } else {
+      return connectivityResult.value != ConnectivityResult.none &&
+        internetConnectionStatus.value == InternetConnectionStatus.connected;
```
Comment (hoangdat) [inline lib/features/network_connection/presentation/network_connection_controller.dart:108]: IMO, `internetConnectionStatus.value == InternetConnectionStatus.connected;` is also a network connected

Comment (dab246) [reply lib/features/network_connection/presentation/network_connection_controller.dart:108]: I do not understand what you mean?

**Review by hoangdat — APPROVED** [review]

**Pull request merged**